### PR TITLE
[prometheus-cloudwatch-exporter] Add extraEnv to values file to specify extra env vars being added to the deployment

### PR DESCRIPTION
**Pull request by AClarkie**
#### What this PR does / why we need it

This PR adds `extraEnv` as an option in the values.yaml which can be used to add extra environment variables to the deployment. To simplify the `env` logic I've added `POD_IP` as an env var that will always be present.

The reason it's needed is that I want the option to set an environment variable `AWS_SHARED_CREDENTIALS_FILE` to instruct where the aws credentials file exists on disk, so that I can use the vault injector to mount the creds to a specific directory.

#### Special notes for your reviewer

I'm happy to modify the logic if required, please let me know

#### Checklist

- [ ] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [ ] Chart Version bumped
- [ ] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Comment by AClarkie** [comment]
@asherf @gianrubio @torstenwalter any chance we can get some eyes on this today?

File: charts/prometheus-cloudwatch-exporter/templates/deployment.yaml [hunk 1]
```diff
@@ -39,9 +39,16 @@ spec:
...
       containers:
         - name: {{ .Chart.Name }}
+          env:
+            - name: POD_IP
+              valueFrom:
+                fieldRef:
+                  fieldPath: status.podIP
+        {{- if .Values.extraEnv }}
+        {{- .Values.extraEnv  | toYaml | nindent 12 }}
+        {{- end }}
```
Comment (zanhsieh) [inline charts/prometheus-cloudwatch-exporter/templates/deployment.yaml:49]: This should be better to deal with empty object `{}`:

```suggestion
        {{- with .Values.extraEnv }}
        {{- toYaml . | nindent 12 }}
        {{- end }}
```

**Review by zanhsieh — CHANGES_REQUESTED** [review]

**Review by monotek — CHANGES_REQUESTED** [review]
Why don't you define the pod ip via extraenv? 
Imho it's not needed to be hard coded.

**Comment by AClarkie** [comment]
> Why don't you define the pod ip via extraenv?
Imho it's not needed to be hard coded.

It's purely to simplify the env logic otherwise you need to check if each of the other variables like `.Values.aws.secret.name`, `.Values.aws.aws_secret_access_key` etc. are set before adding in the `env:` which imo would complicate the setup. A very similar approach is taken in the other community charts, see the below examples:

https://github.com/prometheus-community/helm-charts/blob/d7a033184b896d865b76507e585366c86d3cec65/charts/prometheus/templates/alertmanager/deploy.yaml#L53

https://github.com/prometheus-community/helm-charts/blob/d7a033184b896d865b76507e585366c86d3cec65/charts/prometheus/templates/alertmanager/sts.yaml#L52

**Review by monotek — APPROVED** [review]

**Review by monotek — CHANGES_REQUESTED** [review]

**Review by monotek — APPROVED** [review]

**Pull request merged**